decisions/log/0027-footprint-fragment.md

```diff
@@ -0,0 +1,112 @@
+3. The requested host system will attempt to always send a response event to the requesting host system, **but the requesting host system should not rely on this behavior and retry after an appropriate amount of time by sending another new request event.**
+
+### Async Event Processing: Business Case 1 – Requesting Product Footprints
+
+#### Context and Assumptions
+
+- This business case is triggered by a data recipient if they wish to access a Product Footprint **they could not yet access through the `ListFootprints` API.**
+- The data recipient and data owner have a mutually agreed upon way to identify products
+
+#### Workflow
+
+1. The data recipient sends a `ProductFootprintFragment`  to the `Events Endpoint` of the data owner. The data recipient should always include 1 or more product ids in property `productIds`. The data recipient should limit each fragment to exactly 1 specific product (e.g. a specific type of apple instead of all apples that somebody is offering)
+    1. If the recipient is requesting a specific reporting period, it should include the reporting period accordingly (example …)
+    2. if the recipient is requesting a specific geography, it should include …
+    3. if the recipient has another need that is not covered above, the data recipient should <…>
+        1. Note: it is possible that this request will not be commonly understood by the data owner’s host system
+
+#### Cases
+
+- Case 0: The solution does not support this kind of processing
+    - either because the event processing does not exit at all, or the product footprint fragment as it is submitted by the data recipient is not supported by the requested host system
+    - The `events` endpoint responds with HTTP error code `400` and with a body with error code `NotImplemented`
+- Case 1: A PCF does not exist (yet) or a partially matching PCF exists
+    - accept the event (HTTP Code 200 returned by the events endpoint)
+    - In case of a partial match, the data owner needs to decide whether to calculate the PCF(s) or not.
+        - Note: the decision making and decision making protocol for this case is up to the discretion of each data owner
+    - In case the data owner decided or needs to calculate the PCF and the calculation succeeded,
+        - the host system makes the newly calculated footprints also available to the data recipient through `ListFootprints`
+        - the host system of the data owner sends back the 1 or more product footprints in a single event to the data requester
+    - in case the data owner decided to not make additional PCFs available
+        - the host system responds listing the (partially) matching PCFs
+    - if the product cannot be found or otherwise identified through the product footprint fragment
+        - the host system of the data owner responds with a PF Response Error Event with code `NoSuchFootprint`
```

Comment:
If the Event was already accepted (line 90), how would you send an error event?

Reply:
I believe the key here is to distinguish between the HTTP response to the PF Request Event and the sending of a PF Response Event. In this case, we have something like:
- The data recipient sends a PF Request Event through the HTTP action Events;
- The data owner responds with HTTP 200, thereby accepting the PF Request Event (i.e., confirming that they will process it);
- The data owner processes the request and concludes that there is no footprint with the requested data;
- The data owner sends PF Response Event with error `NoSuchFootprint`.